File: src/main/java/com/lannstark/lec06/Lec06Main.java
package com.lannstark.lec06;

import java.util.Arrays;
import java.util.List;

public class Lec06Main {

  private static void forEach() {
    List<Long> longs = Arrays.asList(1L, 2L, 3L);
    longs.forEach(System.out::println);
  }

  private static void forLoopDown() {
    List<Long> longs = Arrays.asList(1L, 2L, 3L);
    for(int i = 2; i >= 0; i--) {
      System.out.println(longs.get(i));
    }
  }

  private static void forLoopStep2() {
    for(int i = 0; i <= 5; i += 2) {
      System.out.println(i);
    }
  }



  public static void main(String[] args) {

    for (int i = 1; i <= 5; i += 2) {
      System.out.println(i);
    }

    int i = 1;
    while (i <= 3) {
      System.out.println(i);
      i++;
    }

  }

}

File: src/main/java/com/lannstark/lec06/Lec06Main.kt
package com.lannstark.lec06

fun forEach() {
    val longs = arrayListOf(1L, 2L, 3L)
    longs.forEach(::println)
}

fun forLoopDown() {
    val longs = arrayListOf(1L, 2L, 3L)
    for (i in 2 downTo 0) {
        println(longs[i])
    }
}

fun forLoopStep2() {
    for (i in 0..5 step 2) {
        println(i)
    }
}

fun doWhile() {
    var i = 1
    do {
        println(i)
        i++
    } while (i <= 3)
}
